src/account/account.cairo

```diff
@@ -131,7 +148,8 @@ mod Account {
 
         fn set_public_key(ref self: ContractState, new_public_key: felt252) {
             assert_only_self();
-            self.public_key.write(new_public_key);
+            self.emit(OwnerRemoved { removed_owner_guid: self.public_key.read() });
```

Comment:
why emitting here and not in `_set_public_key`? I'd expect anyone using it as a building block to remain compliant with the event firing

Reply:
ah because of the very first key setting?

Reply:
> ah because of the very first key setting?

Exactly. I considered consolidating as you mentioned and having the event fire `OwnerRemoved` for the zero address upon deployment. It's simpler to implement. If we did that though, why not just consolidate the events themselves into a single event e.g. `OwnerChanged { prev_owner, new_owner }`?

If we want to keep both events, it's better IMO to not fire `OwnerRemoved` upon the initial key setting

Reply:
it can be discussed. a priori not always an owner has to change in order to add a new one, i assume you suggest having `prev_owner == ZERO` when that happens.

Reply:
> a priori not always an owner has to change in order to add a new one, i assume you suggest having prev_owner == ZERO when that happens.

Yes, indeed